src/helper/remote_config/listeners/engine_listener.hpp

```diff
@@ -56,6 +56,7 @@ class engine_listener : public listener_base {
         aggregators_;
     engine::ptr engine_;
     rapidjson::Document ruleset_;
+    std::unordered_set<config_aggregator_base *> seen_;
```

Comment:
nit: I dislike this name. It's not seen its kinda `with_changes`, `to_commit`. I could neither come up with a good name

Reply:
I like `to_commit`.